### PR TITLE
Fix the crossed wires with the Divi classic editor setting.

### PR DESCRIPTION
**Pull request by Camwyn**
[TEC-4088]  

Requires (and includes) https://github.com/the-events-calendar/tribe-common/pull/1695

[TEC-4088]: https://theeventscalendar.atlassian.net/browse/TEC-4088?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -657,6 +657,7 @@ public function bind_implementations() {
 		tribe_register_provider( Tribe\Service_Providers\Onboarding::class );
 		tribe_register_provider( Tribe\Admin\Notice\Service_Provider::class );
 		tribe_register_provider( Tribe\Admin\Conditional_Content\Service_Provider::class );
+		$bar = '';
```
Comment (bordoni) [inline src/Tribe/Main.php:660]: debug remove.

File: src/Tribe/Editor/Compatibility/Divi.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			}
+		}
+
+		$foo = '';
+		add_filter( 'tribe_editor_should_load_blocks', [ $this, 'filter_tribe_editor_should_load_blocks' ], 20 );
+	}
+
+	public static function is_divi_active() {
+		$theme = wp_get_theme(); // gets the current theme
+		return 'Divi' == $theme->name || 'Divi' == $theme->template || 'Divi' == $theme->parent_theme;
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility/Divi.php:79]: We shoudl cache this check.

File: src/Tribe/Editor/Compatibility/Divi.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ * Editor Compatibility with Divi theme's builder option.
+ *
+ * @since TBD
+ */
+class Divi {
+	public static $classic_key = 'et_enable_classic_editor';
+
+	public static $classic_on = 'on';
+
+	public static $classic_off = 'off';
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility/Divi.php:17]: Docblocks.

File: src/Tribe/Editor/Compatibility/Divi.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+	/**
+	 * Registers the hooks and filters required based on if the Classic Editor plugin is active.
+	 *
+	 * @since TBD
+	 */
+	public function init() {
+		if ( static::is_divi_active() ) {
+			$this->hooks();
+		}
```
Comment (bordoni) [inline src/Tribe/Editor/Compatibility/Divi.php:27]: Invert the check and return.

File: src/Tribe/Settings_Manager.php [hunk 5]
```diff
@@ -163,6 +163,20 @@ public static function set_option( $name, $value ) {
...
+	 *
+	 * @param string $name
+	 *
+	 * @return bool
+	 */
+	public static function remove_option( $name ) {
+		$options          = self::get_options();
+		unset( $options[ $name ] );
+
+		return self::set_options( $options );
```
Comment (bordoni) [inline src/Tribe/Settings_Manager.php:177]: Static?

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Settings_Manager.php (hunk 5 above)
Comment (Camwyn) [reply src/Tribe/Settings_Manager.php:177]: I followed `set_option()` above?

**Review by bordoni — APPROVED** [review]

**Pull request merged**